sdlib/d/gc/slab.d

```diff
@@ -64,7 +64,8 @@ public:
 
 	@property
 	size_t usedCapacity() {
-		return sg.size - freeSpace;
+		// Allocs which do not support metadata have a defined capacity of 0:
+		return _allowsMetaData ? sg.size - freeSpace : 0;
```

Comment:
1. Remeber this is limitation we added for simplicity's sake because we couldn't guarantee that 0 length slice would be handled properly, but not something we really want, just a limitation we accept for the sake of getting things moving.
2. is not a significant change in behavior: in either case you cannot append to the array.